### ISSUE TITLE
`enforce-consistent-class-order` puts leading before text size

### ISSUE THREAD

**Issue by kachkaev** [schoero/eslint-plugin-better-tailwindcss#145]
👋 @schoero! Great plugin! I'm switching from `eslint-plugin-tailwindcss`, things feel great so far!

There is one small issue I’ve noticed. It's somewhat to do with `tailwind-merge` - https://github.com/dcastil/tailwind-merge/issues/573#issuecomment-3019164793

When `leading-*` is used next to `text-*`, usage of arbitrary values in `text-*` affects the order of class names:

```ts
/* eslint better-tailwindcss/enforce-consistent-class-order: "warn" */
const examples = [
  // this order is expected (text-xl and text-xl/5 has both font size and line height)
  cn("text-xl leading-3"),
  cn("text-xl leading-[2]"),
  cn("text-xl leading-[calc(1+1)]"),
  cn("text-xl/5 leading-3"),
  cn("text-xl/5 leading-[2]"),
  cn("text-xl/5 leading-[calc(1+1)]"),

  // this order is expected (leading value is invalid; class name is not recognised)
  cn("leading-oops text-xl"),

  // this order is unexpected (leading-* should be able to override text-*)
  cn("leading-3 text-[length:20px]"),
  cn("leading-[2] text-[length:20px]"),
  cn("leading-[calc(1+1)] text-[length:20px]"),

  cn("leading-3 text-[length:20px]/5"),
  cn("leading-[2] text-[length:20px]/5"),
  cn("leading-[calc(1+1)] text-[length:20px]/5"),

  cn("leading-3 text-(length:--some-var)"),
  cn("leading-[2] text-(length:--some-var)"),
  cn("leading-[calc(1+1)] text-(length:--some-var)"),

  cn("leading-3 text-(length:--some-var)/5"),
  cn("leading-[2] text-(length:--some-var)/5"),
  cn("leading-[calc(1+1)] text-(length:--some-var)/5"),
];
```

As a result, `leading-*` is erased by `tailwind-merge` because it goes before `text-*`. I believe that `leading-*` should always follow `text-[LENGTH]` to avoid this pitfall. It configures a subset of what `text-[LENGTH]` does, so its relationship to `leading-[VALUE]` is similar to how `p-N` relates to `pt-N`.

In the meantime, I was able to switch to `text-[LENGTH]/[LEADING]`. This might not work in all cases though.

**Comment by schoero** [comment]
Thank you for the detailed issue. By default the `order` of the [enforce-consistent-class-order](https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/main/docs/rules/enforce-consistent-class-order.md#order) rule is set to `"improved"`.

This is basically the official sorting order of tailwindcss but sorts the classes more strictly by their variants. This can make it more readable, especially in combination with the [`enforce-consistent-line-wrapping`](https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/main/docs/rules/enforce-consistent-line-wrapping.md#better-tailwindcssenforce-consistent-line-wrapping) rule.

I had a very simple regex to extract the variants, which caused `text-[length:20px]` to be interpreted as a variant because of the `:`. 

This is now fixed.

If you don't use the `enforce-consistent-line-wrapping` rule, it might be better for you to change the sorting order in `enforce-consistent-class-order` to `"official"`.
